### PR TITLE
WhatsApp API object documentation - MERGE 3/15

### PR DESCRIPTION
**Pull request by timcreasy**
# Pull Request/Issue Resolution

#### Is this change associated with a Braze feature/product release?
- [x] Yes (EA available now, GA on 3/15/23)
- [ ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [x] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [x] Check that all links work.
- [x] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [x] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [x] Tag others as reviewers as necessary.
- [x] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @timothy-kim or @KellieHawks for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs</td>
  </tr>
  <tr>
    <td>@kelliehawks</td>
    <td>Currents<br>Internal Tools<br>Product Partnerships<br>SMS<br>Customer Lifecycle, Identity and Permissions</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Reporting<br>Intelligence<br>User Targeting<br>IAM<br>Channels<br>FIX</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging and Automation<br>Email (Composition and Infrastructure)</td>
  </tr>
</table>
</details>

image

image

image

### CODE REVIEW

File: _docs/_api/objects_filters/messaging/whats_app_object.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+The `whats_app` object allows you to modify or create WhatsApp messages via our [messaging endpoints]({{site.baseurl}}/api/endpoints/messaging).
+
+```json
+{
+    "app_id": (required, string) see App Identifier,
+    "subscription_group_id": (required, string) the id of your subscription group,
+    "message_variation_id": (optional, string) used when providing a campaign_id to specify which message variation this message should be tracked under,
+    "template_name": (required, string) the WhatsApp template name for the message,
+    "template_language_code": (required, string) the language code of the WhatsApp template for the message,
+    "header_variables": (required, object) an object to specify header variable values for specified template_name; see header_variables specification below,
```
Comment (bfalconnier) [inline _docs/_api/objects_filters/messaging/whats_app_object.md:21]: Do we require this even if the template they are specifying has no header variables? It's not optional?
Same for body and button.

File: _docs/_api/objects_filters/messaging/whats_app_object.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "0": "Check it out!",
+    "1": "It's pretty neat."
+}
+```
+
+## button_variables object specification
+
+The `button_variables` object allows you to specify values for button variables in the specified WhatsApp template. Each key is the WhatsApp template variable index to replace with the specified value. **These are zero-indexed**.
+
+**Currently, only one button variable can be specified, which is the path component of a CTA URL. The variable index key must match the CTA URL button index in the message template**
```
Comment (bfalconnier) [inline _docs/_api/objects_filters/messaging/whats_app_object.md:77]: Maybe we should be more direct and say, "if your CTA button is the second button in your template, you must use variable index '1'."  ?

**Review by bfalconnier — APPROVED** [review]

**Review by donniek09 — APPROVED** [review]

**Comment by donniek09** [comment]
Looks good, thanks!

**Review by KellieHawks — APPROVED** [review]

**Comment by KellieHawks** [comment]
Looks great, will merge on 3/15

**Comment by KellieHawks** [comment]
@cla-bot check

**Pull request merged**